### DISCUSSION TITLE
WN 0005 - Records as Dimensions

### DISCUSSION THREAD

**Discussion by lloydtabb** [malloydata/malloy#1424] [What's Next: Language Proposals]
In JSON, Parquet, BigQuery and other data structures nested data can be either singular or repeated.  Malloy today reads singular data but can not create it in a query.  Often, it would be really useful to be able to return data of a known structure for use in other things.  Adding a 'record' function in Malloy that can group dimensional values together and be treated like a single dimension would be useful.

https://github.com/malloydata/whatsnext/blob/main/wns/WN-0005/wn-0005.md

@skokenes @christopherswenson 
@mtoy-googly-moogly @carlineng

**Comment by christopherswenson** [comment]
Maybe instead of function-like syntax, we could use `{}`?

```
dimension: my_record is {
  keep_name,
  new_name is old_name,
  expression is 1 + 1
}
```

**Comment by christopherswenson** [comment]
> Records can't be dereference in a group_by

@lloydtabb Why not?
